ui/raidboss/emulator/data/LogEventHandler.ts

```diff
@@ -0,0 +1,119 @@
+          res.groups.language = i;
+        return res;
+      }
+    }
+    return false;
+  }
+
+  static isMatchStart(line: string): RegExpMatchArray | false {
+    let res;
+    res = LogEventHandler.doesLineMatch(line, EmulatorCommon.countdownRegexes);
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.StartIn = (parseInt(res.groups.time ?? '0') * 1000).toString();
+      res.groups.StartType = 'Countdown';
+      return res;
+    }
+    res = LogEventHandler.doesLineMatch(line, EmulatorCommon.sealRegexes);
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.StartIn = '0';
+      res.groups.StartType = 'Seal';
+      return res;
+    }
+    res = LogEventHandler.doesLineMatch(line, EmulatorCommon.engageRegexes);
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.StartIn = '0';
+      res.groups.StartType = 'Engage';
+      return res;
+    }
+    return false;
+  }
+
+  static isMatchEnd(line: string): RegExpMatchArray | false {
+    let res;
+    res = LogEventHandler.doesLineMatch(line, { '': EmulatorCommon.winRegex });
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.EndType = 'Win';
+      return res;
+    }
+    res = LogEventHandler.doesLineMatch(line, { '': EmulatorCommon.wipeRegex });
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.EndType = 'Wipe';
+      return res;
+    }
+    res = LogEventHandler.doesLineMatch(line, { '': EmulatorCommon.cactbotWipeRegex });
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.EndType = 'Cactbot Wipe';
+      return res;
+    }
+    res = LogEventHandler.doesLineMatch(line, EmulatorCommon.unsealRegexes);
+    if (res) {
+      res.groups = res.groups ?? {};
+      res.groups.EndType = 'Unseal';
+      return res;
+    }
+    return false;
+  }
+
+  public currentFight: LineEvent[] = [];
+  public currentZoneName?: string;
+  public currentZoneId = -1;
+
+  parseLogs(logs: LineEvent[]): void {
+    for (let i = 0; i < logs.length; ++i) {
+      const lineObj = logs[i] as LineEvent;
+
+      this.currentFight.push(lineObj);
+
+      lineObj.offset = lineObj.timestamp - (this.currentFight[0]?.timestamp ?? 0);
+
+      const res = LogEventHandler.isMatchEnd(lineObj.networkLine);
+      if (res) {
+        this.endFight();
+      } else if (lineObj instanceof LineEvent0x01) {
+        const castLineObj = lineObj;
+        this.currentZoneId = parseInt(castLineObj.zoneId ?? '0');
+        this.currentZoneName = castLineObj.zoneName;
+        this.endFight();
+      }
+    }
+  }
+
+  endFight(): void {
+    if (this.currentFight.length < 2)
+      return;
+
+    const start = new Date(this.currentFight[0]?.timestamp ?? 0).toISOString();
+    const end = new Date(this.currentFight.slice(-1)[0]?.timestamp ?? 0).toISOString();
+    this.currentZoneName = this.currentZoneName || 'Unknown';
+    this.currentZoneId = this.currentZoneId || -1;
```

Comment:
```suggestion
    this.currentZoneName = this.currentZoneName ?? 'Unknown';
    this.currentZoneId = this.currentZoneId ?? -1;
```

Alternatively, maybe a `reset()` function for setting all of the `current` variables back to default values initially/after endFight?

Reply:
We explicitly don't want to reset the state to default for LogEventHandler since the ZoneChange line only happens once whereas we might load a bunch of encounters after it for that zone.

I think instead these should just be defaulted in the property declaration? `currentZoneId` is already being defaulted but `currentZoneName` isn't.

Reply:
Ah yeah, that makes sense!

Reply:
Adjusted for next commit.